### PR TITLE
Switch to internal bounding box type.

### PR DESCRIPTION
**Pull request by sayrer**
Another way of fixing #16.

### CODE REVIEW

File: lib/src/boolean/mod.rs [hunk 1]
```diff
@@ -139,3 +139,27 @@ where
         Operation::Union | Operation::Xor => MultiPolygon(subject.iter().chain(clipping).cloned().collect()),
     }
 }
+
+/// A bounded 2D quadrilateral whose area is defined by minimum and maximum `Coordinates`.
+///
+/// A simple implementation copied from geo_types 0.4.0, because this version is a better
+/// fit for the needs of this crate than the newer ones.
+#[derive(PartialEq, Clone, Copy, Debug)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
```
Comment (bluenote10) [inline lib/src/boolean/mod.rs:148]: We can probably get rid of the serde serialization stuff.

**Comment by bluenote10** [comment]
That's basically what I had in mind. Minor notes:

- We probably should bump the version to 0.3 to satisfy semver.
- Perhaps I wouldn't have placed the helper type directly in `mod.rs` but maybe its own module or somewhere in the helpers or so.

**Comment by sayrer** [comment]
@bluenote10 is this good enough to merge?

**Comment by bluenote10** [comment]
Yes, this looks good to me, thanks for the changes. I just don't have merge rights, let's ping @untoldwind.

**Pull request merged**